packages/babel-parser/src/parser/expression.js

```diff
@@ -2049,7 +2057,12 @@ export default class ExpressionParser extends LValParser {
     trailingCommaPos: ?number,
   ): N.ArrowFunctionExpression {
     this.scope.enter(SCOPE_FUNCTION | SCOPE_ARROW);
-    this.prodParam.enter(functionFlags(isAsync, false));
+    let flags = functionFlags(isAsync, false);
```

Comment:
We need to inherit `[In]` when the arrow function body is `{`-less (concise body).